src/components/apps/details/Drawer.js

```diff
@@ -0,0 +1,267 @@
+/**
+ * @author sriram
+ *
+ * A drawer that display tabs with various information such as description, rating, documentation, quick launches about an app.
+ *
+ */
+
+import React, { useState } from "react";
+
+import ids from "../../apps/ids";
+import DETabPanel from "../../utils/DETabPanel";
+import messages from "../../apps/messages";
+import DetailsPanel from "./DetailsPanel";
+import constants from "../../../constants";
+
+import {
+    build,
+    formatMessage,
+    getMessage,
+    Rate,
+    withI18N,
+} from "@cyverse-de/ui-lib";
+import {
+    CircularProgress,
+    Drawer,
+    IconButton,
+    Tab,
+    Tabs,
+    Tooltip,
+    Typography,
+} from "@material-ui/core";
+import { makeStyles } from "@material-ui/core/styles";
+import ToolsUsedPanel from "./ToolUsedPanel";
+import AppFavorite from "../AppFavorite";
+import { injectIntl } from "react-intl";
+import LinkIcon from "@material-ui/icons/Link";
+
+const TABS = {
+    appInfo: "APP INFORMATION",
+    toolInfo: "TOOL(S) USED BY THIS APP",
+};
+const useStyles = makeStyles((theme) => ({
+    drawerPaper: {
+        [theme.breakpoints.up("lg")]: {
+            maxWidth: "25%",
+        },
+        [theme.breakpoints.down("lg")]: {
+            maxWidth: "50%",
+        },
+        [theme.breakpoints.down("sm")]: {
+            maxWidth: "90%",
+        },
+    },
+
+    drawerHeader: {
+        margin: theme.spacing(1),
+        display: "flex",
+        flexDirection: "row",
+        maxWidth: "100%",
+    },
+    drawerSubHeader: {
+        marginLeft: theme.spacing(2),
+        display: "flex",
+        flexDirection: "row",
+        maxWidth: "100%",
+    },
+
+    tabIndicator: {
+        backgroundColor: theme.palette.secondary.main,
+    },
+    tabSelected: {
+        color: theme.palette.primary.contrastText,
+        backgroundColor: theme.palette.primary.main,
+    },
+    headerOperations: {
+        marginLeft: theme.spacing(1),
+    },
+}));
+
+function DetailsHeader({
+    loading,
+    appName,
+    isExternal,
+    isPublic,
+    isFavorite,
+    onFavoriteClick,
+    intl,
+    classes,
+    baseId,
+}) {
+    return (
+        <>
+            <Typography variant="h6" component="span">
+                {appName}
+            </Typography>
+            {!isExternal && isPublic && (
+                <div className={classes.headerOperations}>
+                    <AppFavorite
+                        intl={intl}
+                        baseId={baseId}
+                        isFavorite={isFavorite}
+                        isExternal={isExternal}
+                        onFavoriteClick={onFavoriteClick}
+                    />
+                    {/*TODO: Sriram - implement app link*/}
```

Comment:
We should also have `badge` code for app URLs.
So users can create a badge that links directly to the app without having to create a Quick Launch with no custom params first.